examples/chatgpt/rag-quickstart/azure/Azure_AI_Search_with_Azure_Functions_and_GPT_Actions_in_ChatGPT.ipynb

```diff
@@ -0,0 +1,1330 @@
+                "    \"OPENAI_API_KEY\": os.getenv(\"OPENAI_API_KEY\"),\n",
+                "    \"SEARCH_SERVICE_API_KEY\": os.getenv(\"SEARCH_SERVICE_API_KEY\"),\n",
+                "    \"EMBEDDINGS_MODEL\": os.getenv(\"EMBEDDINGS_MODEL\")\n",
+                "}\n",
+                "\n",
+                "# Create the settings argument for the az functionapp create command\n",
+                "settings_args = []\n",
+                "for key, value in env_vars.items():\n",
+                "    settings_args.append(f\"{key}={value}\")\n",
+                "\n",
+                "subprocess.run([\n",
+                "    \"az\", \"functionapp\", \"config\", \"appsettings\", \"set\",\n",
+                "    \"--name\", app_name,\n",
+                "    \"--resource-group\", resource_group,\n",
+                "    \"--settings\", *settings_args\n",
+                "], check=True)"
+            ]
+        },
+        {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "Now that the function app is created, you are ready to publish your function! This may take up to 10 minutes to deploy"
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "subprocess.run([\n",
+                "    \"func\", \"azure\", \"functionapp\", \"publish\", app_name\n",
+                "], check=True)"
+            ]
+        },
+        {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "## Input into a GPT\n",
+                "Now that we have an Azure Function that queries this Vector Search Index, let's put it as a GPT action!\n",
+                "\n",
+                "See documentation [here](https://openai.com/index/introducing-gpts/) on GPTs and [here](https://platform.openai.com/docs/actions) on GPT Actions. Use the below as the instructions for the GPT and as the OpenAPI spec for the GPT Action.\n"
+            ]
+        },
+        {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "### 1. Generate OpenAPI Spec\n",
+                "Below is a sample OpenAPI spec. Just run the function and it should be copied to your clipboard to paste in the GPT Action! \n",
+                "\n",
+                "Note that this does not have any authentication by default, but you can set up Azure Functions with OAuth by following the pattern in this cookbook in the Authentication section. \n",
+                "\n",
+                "TO-DO: link the SharePoint cookbook"
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "def generate_openapi_spec(app_name):\n",
```

Comment:
probably don't need a function for this; could we just give this to them and tell them to copy it?